src/System.Management.Automation/engine/parser/tokenizer.cs

```diff
@@ -1229,23 +1233,144 @@ private int SkipBlockComment(int i)
             return i;
         }
 
-        private static char Backtick(char c)
+        private string Backtick(char c)
         {
             switch (c)
             {
-                case '0': return '\0';
-                case 'a': return '\a';
-                case 'b': return '\b';
-                case 'e': return '\u001b';
-                case 'f': return '\f';
-                case 'n': return '\n';
-                case 'r': return '\r';
-                case 't': return '\t';
-                case 'v': return '\v';
-                default: return c;
+                case '0': return "\0";
+                case 'a': return "\a";
+                case 'b': return "\b";
+                case 'e': return "\u001b";
+                case 'f': return "\f";
+                case 'n': return "\n";
+                case 'r': return "\r";
+                case 't': return "\t";
+                case 'u': return ScanUnicodeEscapeSequence();
+                case 'v': return "\v";
+                default: return c.ToString();
             }
         }
 
+        private string ScanUnicodeEscapeSequence()
+        {
+            int escSeqStartIndex = _currentIndex - 2;
+            bool isBracketedSequence = false;
+            bool isTerminated = false;
+
+            char c = PeekChar();
+            if (c == '{')
+            {
+                SkipChar();
+                isBracketedSequence = true;
+            }
+
+            int maxNumberOfHexDigits = (isBracketedSequence ? 6 : 4);
+
+            // Scan hex chars after the Unicode escape sequence start.
+            var sb = GetStringBuilder();
+            int i;
+            for (i = 0; i < maxNumberOfHexDigits; i++)
```

Comment:
Ah, maybe we are talking across one another?  When I think of supporting a variable digit count, I'm assuming we only need one form `` "`u{...}"`` where `...` can be one to six hex digits.  If we only supported one form, this should be it.  

The other form is necessarily fixed digit because we want to allow non-Unicode esc seq digits to be right next to the esc seq e.g. this should allowed `` "`u00A9Acme Inc"`` should result in `"@Acme Inc"`.  In my mind, that means the form without a "terminator" is a fixed-width of 4 hex digits - just like it is in [C#](https://msdn.microsoft.com/en-us/library/aa664669(v=vs.71).aspx).  The C# equivalent of the previous example would be `` "\u00A9Acme Inc"``.  Ditto for Bash.